blockchain/state/statedb.go

```diff
@@ -1044,7 +1053,7 @@ func (s *StateDB) Commit(deleteEmptyObjects bool) (root common.Hash, err error)
 			// and just mark it for deletion in the trie.
 			s.deleteStateObject(stateObject)
 		case isDirty:
-			if stateObject.IsProgramAccount() {
+			if stateObject.IsProgramAccount() && stateObject.account.Type() == account.SmartContractAccountType {
```

Comment:
Just use `IsContractAccount`

Reply:
Why did you add this comparison? The storage of EOA can't be set due to this.

Reply:
@ulbqb Sorry, This is the code I added before when I wasn't sure if it was the cause of the test failure.
I have removed it from the latest commit.